core/src/modules/locks.ts

```diff
@@ -41,3 +41,35 @@ export async function waitForLock(
 
   return { releaseLock, attempts };
 }
+
+export type getLockResponse = {
+  releaseLock: Function;
+  isLocked: boolean;
+  lockedBy: string;
+};
+
+export async function getLock(
+  key: string,
+  requestId: string = uuid.v4(),
+  ttl = LOCK_DURATION_MS,
+  lockType?: string
+): Promise<getLockResponse> {
+  const client = api.redis.clients.client;
+  const lockKey = `grouparoo:lock:${
+    lockType.length > 1 && `${lockType}:`
+  }${key}`;
+
+  const set = await client.setnx(lockKey, requestId);
+
+  await client.expire(lockKey, Math.ceil(ttl / 1000));
+
+  const isLocked = !set;
+
+  async function releaseLock() {
+    await client.del(lockKey);
+  }
+
+  const lockValue = await client.get(lockKey);
```

Comment:
Is the lock value meaningful for the caller?  Perhaps I only care if I was able to get the lock or not